src/common/libflux/plugin.h

```diff
@@ -24,6 +24,8 @@ enum {
     FLUX_PLUGIN_RTLD_DEEPBIND =  8,  /* Load with RTLD_DEEPBIND if available */
 };
 
+int flux_deepbind ();
+
```

Comment:
Does this helper really need to be exported via the public API? I'm not sure if you had a use case for that.

If not, it could possibly go into libutil, or it could get a function name that doesn't start with `flux_` and the prototype in a `plugin_private.h` header. (We've done that with some other APIs in libflux, see e.g. `message_private.h` etc.)

Reply:
It does not. I'm fine with either solution, it's mainly where it is because all the callers already use this header.  Do you have a preference?

Reply:
I do not have a preference. Personally I'd just rename the function and put it in `plugin_private.h` just because that's less work :shrug: If we think there might be more functions like this in the future, we may want to move it to a new `.c` and `.h` in libutil :shrug: (easy to move later though)

Reply:
Pushed an update to rename it, move into a private header, and only read the env var once.  I'm not completely sure this is always called from the same thread, so it's using pthread_once to initialize the value once and then re-use it.  Makes me miss the c++11 "statics are initialized exactly once" guarantee. 😞